lib/spack/spack/compilers/__init__.py

```diff
@@ -576,9 +576,8 @@ def _default(search_paths):
                         )
                         command_arguments.append(detect_version_args)
 
-        # Reverse it here so that the dict creation (last insert wins)
-        # does not spoil the intended precedence.
-        return reversed(command_arguments)
+        # Order already respects priority from compiler class
```

Comment:
Generally when you remove a comment and replace it with a new one, the new comment shouldn't "respond" to the old one. In other words, I would rephrase like:

> Return command arguments using the order specified by the compiler class (i.e. leave them unchanged)